floodsub.go

```diff
@@ -319,6 +365,21 @@ func msgID(pmsg *pb.Message) string {
 	return string(pmsg.GetFrom()) + string(pmsg.GetSeqno())
 }
 
+// validate is called in a goroutine and calls the validate functions of all subs with msg as parameter.
+func (p *PubSub) validate(subs []*Subscription, msg *Message) bool {
+	for _, sub := range subs {
+		ctx, cancel := context.WithTimeout(p.ctx, validateTimeoutMillis*time.Millisecond)
+		defer cancel()
+
+		if sub.validate != nil && !sub.validate(ctx, msg) {
```

Comment:
so the validation timeout is not enforced at all, but simply passed into the validator procedure as the context?